pkg/build/stage/image_spec_mutate.go

```diff
@@ -0,0 +1,214 @@
+package stage
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/werf/common-go/pkg/util"
+	"github.com/werf/werf/v2/pkg/config"
+	"github.com/werf/werf/v2/pkg/container_backend"
+	"github.com/werf/werf/v2/pkg/image"
+	"github.com/werf/werf/v2/pkg/storage"
+)
+
+var ErrImageSpecMutateStageIsNotSupported = fmt.Errorf("error creating image spec mutate stage: can't mutate local image. please provide repo with --repo flag or WERF_REPO env variable")
+
+type ImageSpecMutateStage struct {
+	*BaseStage
+	imageSpecConfig *config.ImageSpec
+}
+
+func GenerateImageSpecMutateStage(imageSpec *config.ImageSpec, baseStageOptions *BaseStageOptions, stagesStorage storage.PrimaryStagesStorage) (*ImageSpecMutateStage, error) {
+	return newImageSpecMutateStage(imageSpec, baseStageOptions, stagesStorage)
+}
+
+func newImageSpecMutateStage(imageSpec *config.ImageSpec, baseStageOptions *BaseStageOptions, stagesStorage storage.PrimaryStagesStorage) (*ImageSpecMutateStage, error) {
+	s := &ImageSpecMutateStage{}
+	switch stagesStorage.(type) {
+	case *storage.RepoStagesStorage:
+		s.imageSpecConfig = imageSpec
+		s.BaseStage = NewBaseStage("imageSpecMutate", baseStageOptions)
```

Comment:
Let’s name it imageSpec and add it to the other stages here
https://github.com/werf/werf/blob/main/pkg/build/stage/base.go#L22-L74